### PR TITLE
usbhost: Can now handle multiple interface descriptors.

### PR DESCRIPTION
**Pull request by ldube**
## Summary
See #3644. Code from:
https://github.com/apache/nuttx/compare/master...btashton:libusb

## Impact
Fix issue #3644

## Testing
Plug in USB mouse+keyboard dongle.
/dev/kbda and /dev/mouse0 can be seen in nsh. Only kbda device was seen before this change.

### CODE REVIEW

**Review by btashton — APPROVED** [review]
Thank you for pulling this code I wrote far too long ago into a PR! I should have done that then.

File: drivers/usbhost/usbhost_enumerate.c [hunk 1]
```diff
@@ -51,7 +51,9 @@ static void usbhost_putle16(uint8_t *dest, uint16_t val);
 static inline int usbhost_devdesc(const struct usb_devdesc_s *devdesc,
               FAR struct usbhost_id_s *id);
 static inline int usbhost_configdesc(const uint8_t *configdesc, int desclen,
-              FAR struct usbhost_id_s *id);
+                                     const uint8_t start_ifnum,
```
Comment (xiaoxiang781216) [inline drivers/usbhost/usbhost_enumerate.c:54]:
```suggestion
                                     uint8_t start_ifnum,
```

File: drivers/usbhost/usbhost_enumerate.c [hunk 2]
```diff
@@ -130,6 +132,8 @@ static inline int usbhost_devdesc(FAR const struct usb_devdesc_s *devdesc,
  ****************************************************************************/
 
 static inline int usbhost_configdesc(const uint8_t *configdesc, int cfglen,
+                                     const uint8_t start_ifnum,
```
Comment (xiaoxiang781216) [inline drivers/usbhost/usbhost_enumerate.c:135]:
```suggestion
                                     uint8_t start_ifnum,
```

File: drivers/usbhost/usbhost_enumerate.c (hunk 1 above)
Comment (ldube) [reply drivers/usbhost/usbhost_enumerate.c:54]: > 
Please give a reason for removing const on what is supposed to be a readonly variable.

Comment (btashton) [reply drivers/usbhost/usbhost_enumerate.c:54]: @xiaoxiang781216  I agree with @ldube, if anything the other parameters here should also have `const`.

Comment (xiaoxiang781216) [reply drivers/usbhost/usbhost_enumerate.c:54]: It's debatable whether the const non pointer parameters should be added const or not, but the whole code base seldom(never?) add const to this type of parameter. I am natural to add const or not for this case, but the same rule must be applied universally to the whole code base. So I would suggest to create a new PR to add const to all const parameter.

Comment (btashton) [reply drivers/usbhost/usbhost_enumerate.c:54]: Sure let's remove const for consistency.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@ldube could you sqaush three patch into one?

**Comment by acassis** [comment]
Thank you @ldube!

**Pull request merged**